R/RLearner_regr_gstat.R

```diff
@@ -0,0 +1,164 @@
+#' @title gstat regression learner.
+#'
+#' @description
+#' mlr learner for regression tasks using [gstat::gstat].
+#'
+#' This learner does not use the krige interface from gstat. This is because we want to make it the most general as possible.
+#' Therefore the learner makes use the combination of gstat::gstat and gstat::predict to compute spatial predidction.
+#' You can read more this StackOverflow thread : https://stackoverflow.com/questions/13920342/how-to-make-ordinary-kriging-by-using-gstat-predict.
+#'
+#' The learner handles gstat variogram autofitting functionnality presented in this post https://www.r-spatial.org/r/2016/02/14/gstat-variogram-fitting.html.
+#' To use manual fitting, you must provide a list to the argument model that holds the following elements :
+#' psill, model, range and nugget.
+#' To use auto fitting, simply provide a list containting the types of models to be tested.
+#' Manual fitting example : lrn.man = makeLearner(cl = 'regr.gstat', id = 'manual', psill = 1, model = 'Sph', range = 900, nugget = 1, locations = ~x+y).\n
+# 'Automatic fitting example : lrn.auto = makeLearner(cl = 'regr.gstat', id = 'auto', psill = c('Sph','Exp','Gau', 'Mat'), locations = ~x+y)"
+#'
+#' @references Edzer J Pebesma
+#' Multivariable geostatistics in S: the gstat package
+#' Computers & Geosciences Volume 30, Issue 7, 2004, 683-691.
+#'
+#'
+#' @name regr.gstat
+#' @rdname regr.gstat
+NULL
+
+#' @export
+makeRLearner.regr.gstat = function() {
+  makeRLearnerRegr(
+    cl = "regr.gstat",
+    package = "gstat",
+    par.set = makeParamSet(
+      # gstat::gstat params
+      makeFunctionLearnerParam(id = "g"),
+      makeUntypedLearnerParam(id = "id"), # FIXME what should be the type ?
+      makeUntypedLearnerParam(id = "locations", default = ~x+y), # FIXME what should be the type ?
```

Comment:
See above. And what are the `x` and `y` here?

Reply:
I assume coordinates? If so, they could be extracted from the task where one can set argument `coordinates` to make it a spatial task. Finally a good use case ;)

Reply:
Yes, `x` and `y` are indeed coordinates expressed as a formula. In this case should the I use ` makeFunctionLearnerParam` rather than `makeUntypedLearnerParam` ?

Reply:
I would be tempted not to make this parameters at all and instead say in the documentation that coordinates always need to be x and y.

Reply:
Alright, I'll proceed this way